### ISSUE TITLE
I have some questions I'd like to ask.

### ISSUE THREAD

**Issue by kuaile258890** [penghu-cs/UCCH#14]
Great job. Using the command you provided `python UCCH.py --data_name mirflickr25k_fea --bit 128 --alpha 0.7 --num_hiden_layers 3 2 --margin 0.2 --max_epochs 20 --train_batch_size 256 --shift 0.1 --lr 0.0001 --optimizer Adam`, I was able to achieve nearly similar results (as follows):
Epoch: 19 / 20
[================= 70/70 ====================>] Step: 35ms | Tot: 2s485ms | Loss: 13.143 | LR: 0.0001
Evaluation
Img2Txt: 0.757773 Txt2Img: 0.757578 Avg: 0.757676
Test
Img2Txt: 0.769204 Txt2Img: 0.746224 Avg: 0.757714

However, I noticed that when I start from the Raw data with this command: `python UCCH.py --data_name mirflickr25k --bit 128 --alpha 0.7 --num_hiden_layers 3 2 --margin 0.2 --max_epochs 20 --train_batch_size 256 --shift 0.1 --lr 0.0001 --optimizer Adam --pretrain -a vgg11` for training, the result is only about 0.56. The code you provided does not train the backbone, and I have also tried vgg19 and even a pretrained vit, but the results are all about 0.56. This is a significant difference from the results with the vgg extracted features you provided.

I am wondering if you have done any additional processing on the backbone.

Best regards.

**Comment by penghu-cs** [comment]
Hi,

Thank you for your interest in our work. I apologize for the delayed response. You could try increasing the `warmup_epoch` parameter, since the pretrained backbone might be affected by the gradients generated by the untrained MLPs. For example, you could run the following command:
```python UCCH.py --data_name mirflickr25k --bit 128 --alpha 0.7 --num_hiden_layers 3 2 --margin 0.2 --max_epochs 20 --train_batch_size 256 --shift 0.1 --lr 0.0001 --optimizer Adam --pretrain -a vgg11 --warmup_epoch 5```

Best,
Peng

**Comment by kuaile258890** [comment]
> Hi,
> 
> Thank you for your interest in our work. I apologize for the delayed response. You could try increasing the `warmup_epoch` parameter, since the pretrained backbone might be affected by the gradients generated by the untrained MLPs. For example, you could run the following command: `python UCCH.py --data_name mirflickr25k --bit 128 --alpha 0.7 --num_hiden_layers 3 2 --margin 0.2 --max_epochs 20 --train_batch_size 256 --shift 0.1 --lr 0.0001 --optimizer Adam --pretrain -a vgg11 --warmup_epoch 5`
> 
> Best, Peng

Thank you for your reply. I tried the command you provided, but it did not solve the problem. This issue persists even when I re-download your code to ensure there are no errors in the modifications I make to your code. I even manually modified the code to freeze the backbone, but the result is still only 0.56. In my experiments, I also noticed that there is not much difference between the first epoch and the subsequent epochs. Moreover, I noticed that the warm-up is set to 1 by default. Therefore, even if I use the command you provided earlier, the backbone should be frozen in the first epoch, and its results should not differ much from those using your provided features in the first epoch. However, even in the first epoch, the results using raw data and features are significantly different.

I would like to inquire whether the pretrained weights of the VGG you used for feature extraction are the official ones provided by torchvision, or if they are pretrained weights of VGG provided by other software. Or is it possible that the raw data you provided and the raw data used for VGG extraction are different?

Thank you for your patient response. This is excellent work.

Best regards.

**Comment by penghu-cs** [comment]
Thank you for your interest in my work. I think the problem you encountered is caused by the mismatch between your “mirflickr25k-iall.mat” and “mirflickr25k-yall.mat” files. The “mirflickr25k-yall.mat” file I provided contains shuffled texts, which means that the order of the data is different from the original data in “mirflickr25k-iall.mat”. You need to make sure that the order of images in your “mirflickr25k-iall.mat” is the same as the order of texts in the “mirflickr25k-yall.mat”. 

Best,
Peng

**Comment by penghu-cs** [comment]
Thank you for your interest in our work. I think the problem you encountered is caused by the mismatch between your “mirflickr25k-iall.mat” and “mirflickr25k-yall.mat” files. The “mirflickr25k-yall.mat” file I provided contains shuffled texts, which means that the order of the data is different from the original data in “mirflickr25k-iall.mat”. You need to make sure that the order of images in your “mirflickr25k-iall.mat” is the same as the order of texts in the “mirflickr25k-yall.mat”.

Best,
Peng

**Comment by penghu-cs** [comment]
> > Hi,
> > Thank you for your interest in our work. I apologize for the delayed response. You could try increasing the `warmup_epoch` parameter, since the pretrained backbone might be affected by the gradients generated by the untrained MLPs. For example, you could run the following command: `python UCCH.py --data_name mirflickr25k --bit 128 --alpha 0.7 --num_hiden_layers 3 2 --margin 0.2 --max_epochs 20 --train_batch_size 256 --shift 0.1 --lr 0.0001 --optimizer Adam --pretrain -a vgg11 --warmup_epoch 5`
> > Best, Peng
> 
> Thank you for your reply. I tried the command you provided, but it did not solve the problem. This issue persists even when I re-download your code to ensure there are no errors in the modifications I make to your code. I even manually modified the code to freeze the backbone, but the result is still only 0.56. In my experiments, I also noticed that there is not much difference between the first epoch and the subsequent epochs. Moreover, I noticed that the warm-up is set to 1 by default. Therefore, even if I use the command you provided earlier, the backbone should be frozen in the first epoch, and its results should not differ much from those using your provided features in the first epoch. However, even in the first epoch, the results using raw data and features are significantly different.
> 
> I would like to inquire whether the pretrained weights of the VGG you used for feature extraction are the official ones provided by torchvision, or if they are pretrained weights of VGG provided by other software. Or is it possible that the raw data you provided and the raw data used for VGG extraction are different?
> 
> Thank you for your patient response. This is excellent work.
> 
> Best regards.

Thank you for your interest in our work. I think the problem you encountered is caused by the mismatch between your “mirflickr25k-iall.mat” and “mirflickr25k-yall.mat” files. The “mirflickr25k-yall.mat” file I provided contains shuffled texts, which means that the order of the data is different from the original data in “mirflickr25k-iall.mat”. You need to make sure that the order of images in your “mirflickr25k-iall.mat” is the same as the order of texts in the “mirflickr25k-yall.mat”.

Best,
Peng
